tests/test_version_range.py

```diff
@@ -150,3 +152,19 @@ def test_GemVersionRange_from_native_range_with_pessimistic_operator(self):
             VersionConstraint(comparator=">=", version=RubygemsVersion(string="2.0.8")),
             VersionConstraint(comparator="<", version=RubygemsVersion(string="2.1")),
         )
+
+
+@pytest.mark.parametrize("scheme, range_class", RANGE_CLASS_BY_SCHEMES.items())
+class TestVersionRangeImplementations:
+    VERSION_RANGES_BY_SCHEMES = {
```

Comment:
What about making this what you parametrize on?
And adding a separate, non-parametrized test that fails if there is not one test for every supported/known vers/purl types?